### PR TITLE
(1214a) Risks and needs - Section 2 - Offence analysis page

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

New section required to display Risks and needs

## Changes in this PR
Adds new controller for Risks and needs pages/data and associated routes and template files.

## Screenshots of UI changes

image

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/views/referrals/show/risksAndNeeds/partials/layout.njk [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
```
Comment (jsrobertson) [inline server/views/referrals/show/risksAndNeeds/partials/layout.njk:1]: In an upcoming PR, I've actually replaced this in favour of a rootLayout which can be used by all sub sections under viewing a referral, risks and needs and status history.

File: server/views/referrals/show/risksAndNeeds/offenceAnalysis.njk [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{% extends "./partials/layout.njk" %}
+
+{% block risksAndNeedsContent %}
+  {{ govukInsetText({
+    classes: "govuk-!-margin-top-0",
+    text: importedFromText,
+    attributes: {
+      "data-testid": "imported-from-text"
+    }
+  }) }}
```
Comment (yndajas) [inline server/views/referrals/show/risksAndNeeds/offenceAnalysis.njk:12]: Will this be on every risks and needs page? If so, would it make sense to put it in the layout?

File: server/views/referrals/show/risksAndNeeds/partials/layout.njk [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{% extends "../../../../partials/layout.njk" %}
+
+{% block personBanner %}
+  {% include "../../../_personBanner.njk" %}
+{% endblock personBanner %}
+
+{% block backLink %}
+  {{ govukBackLink({
+    text: "Back to referrals",
+    href: "#"
```
Comment (yndajas) [inline server/views/referrals/show/risksAndNeeds/partials/layout.njk:13]: Should this go to one of the case lists? Maybe determining the destination is a little complicated for now though

File: server/controllers/shared/risksAndNeedsController.test.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      const requestHandler = controller.offenceAnalysis()
+      await requestHandler(request, response, next)
+
+      assertSharedDataServicesAreCalledWithExpectedArguments()
+
+      expect(response.render).toHaveBeenCalledWith('referrals/show/risksAndNeeds/offenceAnalysis', {
+        ...sharedPageData,
+        importedFromText: `Imported from OASys on ${DateUtils.govukFormattedFullDateString()}.`,
+        navigationItems: ShowRisksAndNeedsUtils.viewRisksAndNeedsNavigationItems(request.path, referral.id),
```
Comment (yndajas) [inline server/controllers/shared/risksAndNeedsController.test.ts:79]: Should we mock the util responses here?

**Review by yndajas — COMMENTED** [review]
Just a few small suggestions - looks good in general!

File: server/views/referrals/show/risksAndNeeds/offenceAnalysis.njk (hunk 2 above)
Comment (jsrobertson) [reply server/views/referrals/show/risksAndNeeds/offenceAnalysis.njk:12]: Yeah good shout, I think I'll do this in #440.

File: server/views/referrals/show/risksAndNeeds/partials/layout.njk (hunk 3 above)
Comment (jsrobertson) [reply server/views/referrals/show/risksAndNeeds/partials/layout.njk:13]: Yeah it should, I've just raised card 1269 on the board to keep track of this.

File: server/utils/referrals/showRisksAndNeedsUtils.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import type { Request } from 'express'
+
+import { assessPathBase, assessPaths, referPaths } from '../../paths'
+import type { Referral } from '@accredited-programmes/models'
+import type { MojFrontendNavigationItem } from '@accredited-programmes/ui'
+
+export default class ShowRisksAndNeedsUtils {
+  static viewRisksAndNeedsNavigationItems(
```
[the remark this answers is not in the document]
Comment (jsrobertson) [reply server/utils/referrals/showRisksAndNeedsUtils.ts:8]: I think maybe stick with `ShowRisksAndNeedsUtils` for now but change `viewRisksAndNeedsNavigationItems` to `navigationItems`.  Just thinking ahead to the new utils that are required for the other pages beneath Risks and needs such as Offence analysis etc. Will need a few utils for those pages for the summary list rows etc.

[the remark this answers is not in the document]
Comment (yndajas) [reply server/utils/referrals/showRisksAndNeedsUtils.ts]: Forgot about `ShowReferralUtils`! I think that was more `Show` in contrast to `New` after splitting up the big `ReferralUtils`, whereas risks and needs only show up in `Show`, but yeah, might make sense to use `Show` as a kind of namespacing now

**Review by yndajas — APPROVED** [review]

**Pull request merged**